### PR TITLE
Create 42201.md

### PR DESCRIPTION
**Pull request by johnwight**
Zenip-42201

### CODE REVIEW

**Comment by johnwight** [comment]
Please review @cronicc @ptagl

File: 42201.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+## Specification
+
+**Sidechain Versioning**
+
+In previous zend versions, the `withdrawalEpoch` field was a 4 byte unsigned integer containing the withdrawal epoch length. The maximum valid withdrawal epoch length was limited or capped to 4032.
+In zend 3.1.0, the `withdrawalEpochLength` field is split into two separate fields, as follows.
+1. A 1 byte field for `version`
+2. A 3 byte field for `withdrawalEpochLength`
+             
+
```
Comment (cronicc) [inline 42201.md]: Insert the figure from https://github.com/HorizenOfficial/zen/pull/435#issue-1128409734

File: 42201.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+* new fields added to `getblocktemplate` RPC command
+* new API to retrieve all versions for specified sidechain IDs
+* new value pair added to extend the `getscgenesisinfo` RPC command
+
+
+## Motivation
+
+The enhancements fix an issue with the management of `Custom Fields` included in certificates. They also extend functionality for block template requests. These enhancements have already been implemented with v3.1.0.
+
+The change of Treasury addresses is motivated by the changing global security environment. To increase security, the 3-of-5 multisig signature scheme is changed to 4-of-7.
```
Comment (cronicc) [inline 42201.md]:
```suggestion
The change of Treasury addresses is motivated by the changing global security environment. To increase security, the 3-of-5 multisig signature scheme is changed to 4-of-7. In addition the key holders are now distributed over three separate entities, Zen Blockchain Foundation (ZBF), Horizen Community Council (HCC) and Horizen Labs (HL).
```

**Review by cronicc — CHANGES_REQUESTED** [review]

File: 42201.md (hunk 1 above)
Comment (johnwight) [reply 42201.md]: adding image

**Review by cronicc — DISMISSED** [review]

File: 42201.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+The enhancements fix an issue with the management of `Custom Fields` included in certificates. They also extend functionality for block template requests. These enhancements have already been implemented with v3.1.0.
+
+The change of Treasury addresses is motivated by the changing global security environment. To increase security, the 3-of-5 multisig signature scheme is changed to 4-of-7. In addition the key holders are now distributed over three separate entities: Zen Blockchain Foundation (ZBF), Horizen Community Council (HCC) and Horizen Labs (HL).
+
+## Specification
+
+**Sidechain Versioning**
+
+In previous zend versions, the `withdrawalEpoch` field was a 4 byte unsigned integer containing the withdrawal epoch length. The maximum valid withdrawal epoch length was limited or capped to 4032.
```
Comment (ptagl) [inline 42201.md]: I would like to highlight here that the maximum `WithdrawalEpochLength` is still 4032 after the last update.
```suggestion
In previous zend versions, the `withdrawalEpoch` field was a 4 byte unsigned integer containing the withdrawal epoch length. The maximum valid withdrawal epoch length is limited or capped to 4032.
```

File: 42201.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+*Note: If you construct `coinbasetxn` outside of zend, use `getblockmerkleroots` RPC to calculate these values.*
+
+**Sidechain versioning**
+
+The `GET_SIDECHAINS_VERSIONS` websocket API returns sidechain versions for provided sidechain IDs.
+
+The Sidechain Creation Output now includes sidechain version information. The `sc_create`, `createrawtransaction`, and `getscinfo` RPC commands are updated to include the `version` argument. This field is now mandatory when creating a new sidechain. Also, the `version` field has been added to `CTxScCreationOut` and `ScFixedParamaters`; it is stored in the first byte of `withdrawalEpochLength`. The `sidechainVersion` parameter is added to `GetFieldElement`.
+
+**`getscgensisinfo`extensions**
```
Comment (ptagl) [inline 42201.md]:
```suggestion
**`getscgensisinfo` extensions**
```
Not sure if we have a white space missing here.

File: 42201.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+
+**Block Template Request Extensions**
+
+The `getblocktemplate` method includes two additional return values, based on the optional parameter `roots`:
+ 
+| Key | Required | Type | Description |
+|---|---|---|---|
+| merkleTree | no  | string  | Merkleroot calculated on transactions and certificates |
+| scTxsCommitment | no | string | Calculated on certificates |
```
Comment (ptagl) [inline 42201.md]: The description for both field is quite understandable for me, I just wonder if we could make it clear also for people who are not familiar with Zend.

```suggestion
| merkleTree | no  | string  | Merkle root of the tree containing transactions and certificates of the current block |
| scTxsCommitment | no | string | Merkle root of the sidechain cumulative commitment tree |
```

**Review by ptagl — CHANGES_REQUESTED** [review]

**Review by ptagl — APPROVED** [review]

**Review by cronicc — APPROVED** [review]

**Pull request merged**